opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -259,69 +241,76 @@ public void insert(SQLiteDatabase db, Event event, JSONObject serverJsonObject)
                 event.setFormSubmissionId(generateRandomUUIDString());
             }
             insert(db, Event.class, Table.event, event_column.values(), event, serverJsonObject);
-//            for (Obs o : event.getObs()) {
-//                insert(db, Obs.class, Table.obs, obs_column.values(), obs_column
-// .formSubmissionId.name(), event.getFormSubmissionId(), o, serverJsonObject);
-//            }
         } catch (Exception e) {
             Log.e(getClass().getName(), "", e);
         }
     }
 
-    public long batchInsertClients(JSONArray array) throws Exception {
+    public long batchInsertClients(JSONArray array) {
         if (array == null || array.length() == 0) {
             return 0l;
         }
 
-        long lastServerVersion = 0l;
-
-        getWritableDatabase().beginTransaction();
-
-        for (int i = 0; i < array.length(); i++) {
-            Object o = array.get(i);
-            if (o instanceof JSONObject) {
-                JSONObject jo = (JSONObject) o;
-                Client c = convert(jo, Client.class);
-                if (c != null) {
-                    insert(getWritableDatabase(), c, jo);
-                    if (c.getServerVersion() > 0l) {
-                        lastServerVersion = c.getServerVersion();
+        try {
+            long lastServerVersion = 0l;
+
+            getWritableDatabase().beginTransaction();
+
+            for (int i = 0; i < array.length(); i++) {
+                Object o = array.get(i);
+                if (o instanceof JSONObject) {
+                    JSONObject jo = (JSONObject) o;
```

Comment:
If o is an instance of JSONObject why is this type conversion necessary?

Reply:
o is an Object in line 260 so we need to cast it to a JSONObject